src/box.jl

```diff
@@ -18,6 +18,9 @@ Box(c::SVector{N}, d::SVector{N},
 Box(c::AbstractVector, d::AbstractVector, axes=eye(length(c)), data=nothing) =
     (N = length(c); Box(SVector{N}(c), SVector{N}(d), SMatrix{N,N}(axes), data))
 
+Box(b::NTuple{2,AbstractVector}, axes=eye(length(b[1])), data=nothing) =
```

Comment:
Hmmm...  Not sure if I understood what you mean.  The purpose of the new constructor is to provide a way to create a `Box` out of the two end points `b[1]` and `b[2]` of a solid diagonal.  In this case the center `c` and sides `d` are not immediately available (though they are easy to calculate from `b`, which is what the new constructor does), whereas your `boxcorners` seems to assume availability of `c` and `d`?